service/controller/resource/nodepool/create_deployment_uninitialized.go

```diff
@@ -73,6 +73,10 @@ func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj in
 	if IsNotFound(err) {
 		r.Logger.LogCtx(ctx, "level", "debug", "message", "Azure resource not found, canceling resource")
 		return currentState, nil
+	} else if IsSubnetNotReadyError(err) {
+		r.Logger.LogCtx(ctx, "level", "debug", "message", "subnet is not Ready, it's probably still being created")
+		r.Logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return currentState, nil
```

Comment:
We don't return error since it's an expected case, but we cancel the reconciliation.